src/api/app/components/button_component.rb

```diff
@@ -0,0 +1,49 @@
+class ButtonComponent < ApplicationComponent
+  renders_one :icon
+  renders_one :text
+  renders_one :modal
+
+  TYPES = {
+    danger: 'danger',
+    danger2: 'danger2',
+    warning: 'warning',
+    warning2: 'warning2',
+    success: 'success',
+    success2: 'success2',
+    info: 'info',
+    info2: 'info2'
+  }.freeze
+
+  def initialize(button_type, css_custom_classes: [], button_data: {})
+    super
+
+    @button_type = button_type
+    @css_custom_classes = css_custom_classes
+    @button_data = button_data
+  end
+
+  def css_style
+    css_type = []
+    css_type << case @button_type
```

Comment:
You can simply return directly from the `case` statement. Storing the result of the `case` statement in an array isn't needed since there will always be only one.